ignite/metrics/epoch_metric.py

```diff
@@ -113,7 +116,7 @@ def update(self, output: Tuple[torch.Tensor, torch.Tensor]) -> None:
             except Exception as e:
                 warnings.warn(f"Probably, there can be a problem with `compute_fn`:\n {e}.", EpochMetricWarning)
 
-    def compute(self) -> float:
+    def compute(self) -> Union[int, float, Sequence, Mapping]:
```

Comment:
Let's say we support only sequence or mapping of tensors:
```suggestion
    def compute(self) -> Union[int, float, Sequence[torch.Tensor], Mapping[str, torch.Tensor]]:
```

Reply:
The problem with specifying the type of sequence/mapping is that it will break `mypy` as the `apply_to_type` returns a generic Sequence. 
I am not sure if we really need to update how `apply_to_type`for this usecase.

Reply:
Can't we use `cast` in this case somehow ?

Reply:
Another thing I noticed. There are many test failures because the `precision_recall_curve` use the sklearn compute function which returns a response that doesn't fit with the new validation we setup in this PR i.e. it's not a scalar or a collection of tensors.
Shall I update the implementation of the `precision_recall_curve` to fix that validation? @vfdev-5

Reply:
@Moh-Yakoub it is related to https://github.com/pytorch/ignite/pull/1700 where we tried to fix that. I think what could be done is to comment out failing tests here and uncomment and fix failling tests in #1700 once this PR is merged.
What do you think ?
cc @KickItLikeShika

Reply:
That makes sense. I will add a todo statement for that as well

Reply:
@Moh-Yakoub so, no way to cast the result to `Sequence[torch.Tensor], Mapping[str, torch.Tensor]` here ?

Reply:
@vfdev-5  I've added a type cast, I am not sure if there's any better way to do that cast to allow the `Sequence[torch.Tensor], Mapping[str, torch.Tensor]` return type without annoying mypy.

If you are fine with current approach I will add couple of tests and that PR should be ready.

Reply:
Thanks for the working on that ! I think the best thing we can do here is to add "typing ignore" statement on the output of `apply_to_typefor` and keep the code simple.

Reply:
@Moh-Yakoub I think the issue is that `Sequence` and `Mapping` are from collections and not typing. Please fix it and it will be good to go (think also to update PR to the master).